src/main/scala/cromwell/engine/FinalCall.scala

```diff
@@ -0,0 +1,38 @@
+package cromwell.engine
+
+import cromwell.binding._
+import cromwell.engine.workflow.{ExecutionStoreKey, FinalCallKey}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object FinalCall {
+
+  implicit class FinalCallString(val fqn: FullyQualifiedName) extends AnyVal {
+    /** Does this FQN conform to a final call? */
+    def isFinalCall = fqn startsWith "$final_call"
+    /** Convert this FQN to a `FinalCall` or throw if it can't be converted. */
+    def toStoreKey(workflow: WorkflowDescriptor): ExecutionStoreKey = {
+      fqn match {
+        case CopyWorkflowOutputs.Name => FinalCallKey(CopyWorkflowOutputs(workflow))
+        case _ => throw new IllegalArgumentException(s"Unrecognized call key $fqn")
+      }
+    }
+  }
+}
+
+/** Scope representing a "final call" that is inserted after all other workflow executions. */
+trait FinalCall extends Scope {
+  def execute(implicit ec: ExecutionContext): Future[Unit]
+
+  def prerequisiteCallNames: Set[LocallyQualifiedName] = ???
```

Comment:
Oh sure, you hide behind IntelliJ's automagic when it suits your purpose and not here!!!!

I kid, I don't really have a dog in this fight. I just hate when people throw the actual exception w/ the message "not implemented" :)